bin/oneoff/aichat_requests_to_csv

```diff
@@ -0,0 +1,168 @@
+    ]
+    csv << headers
+
+    student_users_in_section = Section.find(options[:section_id]).followers.map(&:student_user)
+    teacher_users_in_section = Section.find(options[:section_id]).active_section_instructors.map(&:instructor)
+    all_users_in_section = student_users_in_section + teacher_users_in_section
+    rows = AichatRequest.where(
+      user: all_users_in_section,
+    )
+
+    rows.each do |row|
+      user = row.user
+      user_message = JSON.parse(row.new_message)
+      response = row.response
+      user_message_text = user_message["chatMessageText"]
+      execution_status = row.execution_status
+
+      level = Level.find(row.level_id)
+      path = ""
+
+      script_level = level.script_levels.find_by_script_id(row.script_id)
+      if script_level
+        # This handles chat requests that occurred on standard levels in a progression
+        path = script_level.path
+      else
+        # This handles chat requests that occurred on a sublevel
+        parent_levels = BubbleChoice.parent_levels(level.name)
+        parent_levels_in_script = parent_levels.filter do |pl|
+          pl.script_levels.any? {|sl| sl.script_id == row.script_id}
+        end
+
+        if parent_levels_in_script
+          parent_level = parent_levels_in_script.first
+          sublevel_position = parent_level.sublevel_position(level)
+          path = parent_level.build_script_level_path(
+            parent_level.script_levels.first,
+            {sublevel_position: sublevel_position}
+          )
+        end
+      end
+
+      if execution_status == SharedConstants::AI_REQUEST_EXECUTION_STATUS[:USER_PROFANITY]
+        # Parse response for openai/comprehend/blocklist/webpurify details for user message.
+        # There is no model response recorded if user message was flagged.
+        # After https://github.com/code-dot-org/code-dot-org/pull/61095, response was updated to JSON.
+        # Only handle json responses.
+        if valid_json?(response)
+          user_safety_layer_response = JSON.parse(response).deep_symbolize_keys
+          user_blocked_by = user_safety_layer_response[:blocked_by]
+          details = user_safety_layer_response[:details]
+          user_blocked_by_details = get_blocked_by_details(user_blocked_by, details)
+        end
+      elsif execution_status == SharedConstants::AI_REQUEST_EXECUTION_STATUS[:MODEL_PROFANITY]
+        # Model response is recorded in addition to blocked by service details.
+        # After https://github.com/code-dot-org/code-dot-org/pull/61095, response was updated to JSON.
+        # Only handle json responses.
+        if valid_json?(response)
+          model_safety_layer_response = JSON.parse(response).deep_symbolize_keys
+          model_blocked_by = model_safety_layer_response[:blocked_by]
+          details = model_safety_layer_response[:details]
+          model_blocked_by_details = get_blocked_by_details(model_blocked_by, details)
+        end
```

Comment:
minor: looks like these two blocks do the same thing, but just save to different variables. Can we extract to a helper method and/or collapse this condition? We will only ever have safety details for the user or assistant, not both, right?

Reply:
> We will only ever have safety details for the user or assistant, not both, right?
Yes!

Actually, it's only 3 lines since `details` doesn't includ model response so I think I'll keep as-is. Thanks!